File: CarePackage/Software/TelegramDesktop.cs
namespace CarePackage.Software;

public class TelegramDesktop : BaseSoftware
{
    public override string           Key            { get; set; } = "telegram-desktop";
    public override string           Name           { get; set; } = "Telegram";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Social;
    public override string           DownloadName   { get; set; } = "tsetup.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.telegram;
    public override string           Homepage       { get; set; } = "https://telegram.org";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://telegram.org/dl/desktop/win64");
}

File: packages/app/src/services/software/definitions/telegramDesktop.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'telegram-desktop';
	public name = 'Telegram';
	public category = [SoftwareCategory.Social];
	public downloadName = 'tsetup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'telegram.png';
	public homepage = 'https://telegram.org';

	public async resolveDownloadUrl() {
		return ok('https://telegram.org/dl/desktop/win64');
	}
}
